tools/internal/parser/text_test.go

```diff
@@ -9,6 +10,242 @@ import (
 	"github.com/google/go-cmp/cmp"
 )
 
+func TestNormalize(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		in       []byte
+		want     []string
+		wantErrs []error
+	}{
+		{
+			name: "empty_input",
+			in:   []byte{},
+			want: []string{},
+		},
+		{
+			name: "no_early_errors",
+			in: byteLines(
+				"// This is a small replica",
+				"// of the PSL",
+				"com",
+				"net",
+				"lol",
+				"",
+				"// End of file",
+			),
+			want: []string{
+				"// This is a small replica",
+				"// of the PSL",
+				"com",
+				"net",
+				"lol",
+				"",
+				"// End of file",
+			},
+		},
+		{
+			// "utf-16 text"
+			// linewraps are after the spaces
+			name: "utf16be_input_with_bom",
+			in: []byte{
+				0xfe, 0xff, // BOM
+				0x0, 0x75, 0x0, 0x74, 0x0, 0x66, 0x0, 0x2d, 0x0, 0x31, 0x0, 0x36, 0x0, 0x20,
+				0x0, 0x74, 0x0, 0x65, 0x0, 0x78, 0x0, 0x74,
+			},
+			want:     []string{"utf-16 text"},
+			wantErrs: []error{InvalidEncodingError{"UTF-16BE"}},
+		},
+		{
+			name: "utf16le_input_with_bom",
+			// "utf-16 text"
+			// linewraps are after the spaces
+			in: []byte{
+				0xff, 0xfe, // BOM
+				0x75, 0x0, 0x74, 0x0, 0x66, 0x0, 0x2d, 0x0, 0x31, 0x0, 0x36, 0x0, 0x20, 0x0,
+				0x74, 0x0, 0x65, 0x0, 0x78, 0x0, 0x74, 0x0,
+			},
+			want:     []string{"utf-16 text"},
+			wantErrs: []error{InvalidEncodingError{"UTF-16LE"}},
+		},
+		{
+			name: "utf16be_input",
+			// "utf-16 text utf-16 text utf-16 text"
+			// linewraps are after the spaces
+			in: []byte{
```

Comment:
Is there a more readable way to provide these? Maybe by taking the string and encoding it?

Reply:
Yeah good point! I think I was trying to avoid using the same library to both encode and decode the text, in case there are bugs... But that makes no sense, it has its own unit tests and is certainly correct.

Switched to using helpers to have readable input. I also rewrote the malformed UTF-8 test to separate out specific shapes of invalid input and explain what they are.